src/lib/router.js

```diff
@@ -33,6 +33,8 @@ async function swapContent(url) {
     return;
   }
 
+  document.body.classList.add("loading");
```

Comment:
Maybe a dumb question because I haven't thought through the load order but...

Would it be possible to turn these into states in the store (IS_PROGRESSBAR_SHOWING) and put this work into an action and have the router call the action to toggle the state? It might also be good to put `aria-busy=true` on the `<main>` element while we're doing this and set it back to "false" when we're done.

Reply:
Done